NPE in ArcTestRequestScopeProvider.setup() in native mode Fixes #3790

vsevel opened the pull request.

vsevel committed:
NPE in ArcTestRequestScopeProvider.setup() in native mode with dependency on deployment test jar

vsevel commented:
@mkouba @gsmet @geoand CI is green. let me know if you expect anything else on that PR.
thanks

geoand commented:
Thanks @vsevel !

I haven't followed your discussion with @mkouba so I'll let him approve and merge.
One question though, should we perhaps be throwing some exception instead of just logging a warning? Obviously NPE was not good, but perhaps a simple warning isn't enough?

mkouba commented on extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcTestRequestScopeProvider.java:
```diff
@@ -1,21 +1,33 @@
         ArcContainer container = Arc.container();
-        container.requestContext().activate();
+        if (container == null) {
+            LOGGER.warn("ArcTestRequestScopeProvider: container is null, ignoring setup");
```
There is no need to mention `ArcTestRequestScopeProvider` in the log message - it's used as the logger name and will be taken into consideration.

vsevel replied:
@mkouba you are right. initially when I made this defensive code in my vault branch, I looked at the logs:
```
[INFO] Running io.quarkus.it.jpa.postgresql.JPAFunctionalityInGraalITCase
Sep 06, 2019 8:27:20 AM io.quarkus.arc.deployment.ArcTestRequestScopeProvider setup
INFO: container is null, ignoring setup
```
and did not see that the logging line was split on 2 lines (I was searching the expected message without looking above or beyond), and assumed wrongly that there was something special about the native build...

thanks for making the modifications.

mkouba committed:
TestScopeSetup - make it possible to skip action for substrate tests

- ArcTestRequestScopeProvider - skip activation for substrate tests
- resolves #3884

dmlloyd commented:
@stuartwdouglas will you have a chance to take a peek at this?
